lib/src/testing/utils/testrunner.dart

```diff
@@ -746,6 +780,30 @@ void _runTests<PrivateKey, PublicKey>(
       publicKey = pair.publicKey;
       privateKey = pair.privateKey;
     });
+  } else if (c.importKeyException != null) {
+    if (c.privatePkcs8KeyData != null) {
+      test('pkcs8 import exception', () async {
+        try {
+          await r._importPrivatePkcs8Key!(
+              c.privatePkcs8KeyData!, {'curve': 'p-521'});
+          check(false, 'Expected an exception for P-512 import');
+        } catch (e) {
+          check(e.toString().contains(c.importKeyException!));
+        }
+      });
+    }
+    if (c.privateJsonWebKeyData != null) {
+      test('jwk import exception', () async {
+        try {
+          await r._importPrivateJsonWebKey!(
+              c.privateJsonWebKeyData!, {'curve': 'p-521'});
+          check(false, 'Expected an exception for P-512 import');
+        } catch (e) {
+          check(e.toString().contains(c.importKeyException!));
+        }
+      });
+    }
```

Comment:
We need a test case for each kind of key possible:
```
  final List<int>? privateRawKeyData;
  final List<int>? privatePkcs8KeyData;
  final Map<String, dynamic>? privateJsonWebKeyData;
  final List<int>? publicRawKeyData;
  final List<int>? publicSpkiKeyData;
  final Map<String, dynamic>? publicJsonWebKeyData;
```

Also I'd suggest naming the test cases like:
```dart
if (c.publicRawKeyData != null) {
  test('importPublicRawKey() throws', () async {
    ...
  });
}
if (c.publicJsonWebKeyData != null) {
  test('importPublicJsonWebKey() throws', () async {
    ...
  });
}
...

// This test case is testing failure to import a key
// further tests will not be conducted, since they don't
// make sense after failing to import a key.
return;
```